File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<int,int> x,y;
        if(s.size()!=t.size())
            return false;
        for(int i=0;i<s.size();i++){
            x[s[i]]++;
            y[t[i]]++;
        }
        for(int i=0;i<t.size();i++){
            if(x[t[i]]!=y[t[i]])
                return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length())return false;
        Map<Character,Integer> sfreq= new HashMap<>();
        for(char c: s.toCharArray()){
            sfreq.put(c, sfreq.getOrDefault(c, 0 )+1);
        }
        for(char c: t.toCharArray()){
            sfreq.put(c, sfreq.getOrDefault(c, 0 )-1);
            if(sfreq.get(c)==0){
                sfreq.remove(c);
            }
        }
        return sfreq.isEmpty();
    }
}